### PR TITLE
support k up to 2048 in faiss select

### PR DESCRIPTION
**Pull request by benfred**
The faiss repo allows k values up to 2048, but we were limiting to 512 or 1024 instead. This seems to be because of compiler errors such as

```
ptxas error   : Entry function '_ZN4raft7spatial3knn6detail15select_k_kernelIdmLb1ELi2048ELi8ELi64EEEvPKT_PKT0_mmPS4_PS7_S4_S7_i' uses too much shared data (0x10000 bytes, 0xc000 max)
ptxas error   : Entry function '_ZN4raft7spatial3knn6detail15select_k_kernelIdmLb0ELi2048ELi8ELi64EEEvPKT_PKT0_mmPS4_PS7_S4_S7_i' uses too much shared data (0x10000 bytes, 0xc000 max)
ptxas error   : Entry function '_ZN4raft7spatial3knn6detail15select_k_kernelIdmLb1ELi1024ELi8ELi128EEEvPKT_PKT0_mmPS4_PS7_S4_S7_i' uses too much shared data (0x10000 bytes, 0xc000 max)
ptxas error   : Entry function '_ZN4raft7spatial3knn6detail15select_k_kernelIdmLb0ELi1024ELi8ELi128EEEvPKT_PKT0_mmPS4_PS7_S4_S7_i' uses too much shared data (0x10000 bytes, 0xc000 max)
```

when compiling with larger k values.

However, these only affect double precision with 64bit indices - and float32 works up to k=2048 (even with 64 bit indices).

### CODE REVIEW

File: cpp/include/raft/spatial/knn/detail/selection_faiss.cuh [hunk 1]
```diff
@@ -27,10 +27,13 @@ namespace spatial {
...
 constexpr int kFaissMaxK()
 {
-  return (sizeof(key_t) + sizeof(payload_t) > 8) ? 512 : 1024;
+  if (sizeof(key_t) >= 8) {
+    return sizeof(payload_t) >= 8 ? 512: 1024;
+  }
+  return 2048;
 }
 
 template <typename key_t, typename payload_t, bool select_min, int warp_q, int thread_q, int tpb>
```
Comment (cjnolet) [inline cpp/include/raft/spatial/knn/detail/selection_faiss.cuh:39]: Just for consistency could we make this one start with payload_t as well?

**Review by cjnolet — APPROVED** [review]
Looks good, just a small nitpick but I'm pre-approving

**Review by cjnolet — APPROVED** [review]
Github seems to be in a weird state

**Review by ajschmidt8 — APPROVED** [review]
Approving `ops-codeowner` file changes

**Review by cjnolet — APPROVED** [review]

**Pull request merged**